probe-rs/src/architecture/riscv/dtm/jtag_dtm.rs

```diff
@@ -66,18 +67,97 @@ impl Dtm {
         })
     }
 
-    pub fn target_reset_deassert(&mut self) -> Result<(), DebugProbeError> {
-        self.probe.target_reset_deassert()
+    fn transform_dmi_result(response_bytes: Vec<u8>) -> Result<u32, DmiOperationStatus> {
+        let response_value: u128 = response_bytes.iter().enumerate().fold(0, |acc, elem| {
+            let (byte_offset, value) = elem;
+            acc + ((*value as u128) << (8 * byte_offset))
+        });
+
+        // Verify that the transfer was ok
+        let op = (response_value & DMI_OP_MASK) as u8;
+
+        if op != 0 {
+            return Err(DmiOperationStatus::parse(op).expect("INVALID DMI OP status"));
```

Comment:
I'm a bit worried that we will panic here, in my experience hardware is not very good at follwing specs. If this can never panic due to to `DMI_OP_MASK`, please add a comment indicating this.

Reply:
I added a comment, as we mask the op via DMI_OP_MASK it should be impossible barring cosmic rays for `op` to be more than 3